### ISSUE TITLE
Training with layer offloading became 2 to 5 times slower after latest update

### ISSUE THREAD

**Issue by Lexxxco** [ostris/ai-toolkit#848]
## This is for bugs only

Did you already ask [in the discord](https://discord.gg/VXmU2f5WEU)?

Yes

You verified that this is a bug and not a feature request or question by asking [in the discord](https://discord.gg/VXmU2f5WEU)?

Yes

## Training Lora with layer offloading became 2 to 5+ times slower after latest update (main@68e9b38). 

Even when VRAM is not fully loaded (50%), training slowed down drastically. On both Flux 2 dev, and Klein 9B (W10/11)  RTX 6000/5090 - with different python and torch versions (Torch 2.7/2.7.1/2.8 CUDA 12.8 and all torch versions with CUDA 13.0). With LOW VRAM enabled AND disabled. (less noticeable with LOW VRAM disabled).

Most noticeable with Automagic optimizer (up to 5 times slower) with batch >1 and gradient accumulation. For example 32-62 s/it  vs 140-200s/it after update on identical setup (for batch 2x4 and 1-2MP bucket with Flux Klein), less noticeable with ADAM (up to 2 times slower), it is the major slowdown and big problem.  

Issue was not present in older commits ~week ago (main@6102370)

**Comment by PlagueKind** [comment]
Having a somewhat similar situation on blackwell 5060ti 16gb. Across the board. Z-image  kelin , l2p.
Wasn't present before.
8bit optmizers and even automagic2 are not saving vram at all.
Training speed is also slower.
Previously i could run Z-image unquantized and i had no layer offloading and speed was about 1it/s - 1.1s/it.
Now i oom and have to offload, and speed is doubled.

Having a conversation in this moment with someone on the discord with exact same card and he has the same exact identical issue as me.
